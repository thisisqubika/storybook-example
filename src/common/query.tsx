import {
  QueryClient, QueryClientConfig, QueryClientProvider,
} from '@tanstack/react-query';

export function queryClientConfig(
  config?: QueryClientConfig,
): QueryClientConfig {
  return {
    ...config,
    defaultOptions: {
      ...config?.defaultOptions,
      queries: {
        refetchOnWindowFocus: false,
        retry: 1,
        ...config?.defaultOptions?.queries,
      },
    },
  };
} // function newQueryClient

export interface QueryClientWrapperProps {
  config?: QueryClientConfig;
  children: JSX.Element;
} // interface ThemeWrapperProps

export function QueryClientWrapper({
  config,
  children,
}: QueryClientWrapperProps) {
  const queryClient = new QueryClient(queryClientConfig(config));
  return (
    <QueryClientProvider client={queryClient}>
      {children}
    </QueryClientProvider>
  );
} // function QueryClientWrapper

