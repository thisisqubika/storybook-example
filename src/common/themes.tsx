// Check out <https://zenoo.github.io/mui-theme-creator/>.
import {
  createTheme, CssBaseline, ThemeProvider, ThemeOptions,
} from '@mui/material';

export const themeOptions: ThemeOptions = {
  palette: {
    mode: 'light',
    primary: {
      main: '#812020',
    },
    secondary: {
      main: '#cc8333',
    },
    background: {
      default: '#e0cab8',
      paper: '#efe8e2',
    },
    error: {
      main: '#fb3636',
    },
    info: {
      main: '#025e90',
    },
    warning: {
      main: '#e49e01',
    },
    success: {
      main: '#2b9430',
    },
  },
  typography: {
    fontFamily: 'Lora',
    fontSize: 13,
  },
}; // const themeOptions

export interface ThemeWrapperProps {
  children: JSX.Element;
} // interface ThemeWrapperProps

export function ThemeWrapper({
  children,
}: ThemeWrapperProps) {
  return (
    <ThemeProvider theme={createTheme(themeOptions)}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  );
} // function ThemeWrapper
