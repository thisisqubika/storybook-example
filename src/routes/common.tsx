import { isRouteErrorResponse, useRouteError } from 'react-router-dom';

export const PATHS = {
  books: 'books',
}; // const PATHS

export function ErrorBoundary() {
  const error: unknown = useRouteError();
  return isRouteErrorResponse(error) ? (
    <h1>{error.status} {error.statusText}</h1>
  ) : (
    <h1>{(error as Error)?.message ?? error}</h1>
  );
} // function ErrorBoundary
