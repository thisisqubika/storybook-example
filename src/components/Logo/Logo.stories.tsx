import type { Meta, StoryObj } from '@storybook/react';
import { ThemeWrapper } from '../../common/themes';
import { Logo } from '.';

export default {
  component: Logo,
} satisfies Meta; // meta

export const Base: StoryObj = {
  render() {
    return (
      <ThemeWrapper>
        <Logo />
      </ThemeWrapper>
    );
  },
}; // const Base
