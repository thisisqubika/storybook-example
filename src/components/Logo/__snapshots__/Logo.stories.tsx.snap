// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Logo Base smoke-test 1`] = `
<h3 class="MuiTypography-root MuiTypography-h3 css-13bgm19-MuiTypography-root"
    data-testid="Logo"
>
  Books
</h3>
`;
